cityofaustin/atd-moped
# Add 4th layer to the Moped AGOL resource to house exploded multipoint geometry as points

# Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/17999

# Testing

* Get on this branch and replicate the production DB to your machine
* Make yourself a `env_file` or ask me to `wormhole` you one.
   * Thanks to @mddilley for observing that those with existing env files may have to update this line as follows:
```
HASURA_ENDPOINT=http://host.docker.internal:8080/v1/graphql
```
* `cd moped-etl/arcgis/`
* `docker compose run arcgis;`
  * This will run with the default argument, a sole `-d`, which means update the existing data with changes from the last seven days.
* `docker compose run arcgis -d 2024-08-24T18:33:51.026046+00:00` 
  * This will do a similar data update, handling changes since the supplied datetime.
* `docker compose run arcgis -f`
  * This is a full replace. This is painful, as it's more likely than not that you'll encounter an AGOL error, on their end. When this happens, you just have to try again. And you should try again until it works so the data is restored to the correct state.  It took me maybe 10 tries over the course of half an hour to finally get through a full replace, but it did do it eventually.
    * Update on Sept 10: I just ran this doing a full replace and got it first try, so YMMV for the better. 🤞 
* Review the data on AGOL: https://austin.maps.arcgis.com/home/item.html?id=1c084c8756a84e6db7e2796c98c850a2

# Thanks

Thanks to @mddilley for his help on this one -- brainstorming and rubber ducking. 🙏 🦆 🙏 

Also thanks to @johnclary for his input and request to stick with our tried and true patterns. 💯 

---
#### Ship list
~- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~
